implementations/rust/vault/src/software/mod.rs

```diff
@@ -63,37 +88,43 @@ impl DefaultVault {
 
     fn hkdf_sha256_internal(
         &mut self,
-        salt: SecretKeyContext,
+        salt: &Box<dyn Secret>,
         info: &[u8],
         ikm: &[u8],
         output_attributes: Vec<SecretKeyAttributes>,
-    ) -> Result<Vec<SecretKeyContext>, VaultFailError> {
+    ) -> Result<Vec<Box<dyn Secret>>, VaultFailError> {
         let salt = self.get_entry(salt, VaultFailErrorKind::Ecdh)?;
 
         // FIXME: Doesn't work for secrets with size more than 32 bytes
         let okm_len = output_attributes.len() * 32;
```

Comment:
Either we should throw an error if more than 32 bytes is asked or handle it under the hood with multiple calls.